client/dashboard/app/queries/site-plans.ts

```diff
@@ -1,11 +1,31 @@
 import { mutationOptions, queryOptions } from '@tanstack/react-query';
 import { restoreSitePlanSoftware } from '../../data/site-hosting';
-import { fetchCurrentSitePlan } from '../../data/site-plans';
+import { fetchSitePlans } from '../../data/site-plans';
 
 export const siteCurrentPlanQuery = ( siteId: number ) =>
 	queryOptions( {
 		queryKey: [ 'site', siteId, 'plans', 'current' ],
-		queryFn: () => fetchCurrentSitePlan( siteId ),
+		queryFn: async () => {
+			const plans = await fetchSitePlans( siteId );
+			const plan = Object.values( plans ).find( ( plan ) => plan.current_plan );
+			if ( ! plan ) {
+				throw new Error( 'No current plan found' );
+			}
+			return plan;
+		},
+	} );
+
+export const sitePlanBySlugQuery = ( siteId: number, productSlug: string ) =>
+	queryOptions( {
+		queryKey: [ 'site', siteId, 'plans', productSlug ],
+		queryFn: async () => {
```

Comment:
@arthur791004 I think we can still keep the query key and data shared. We just need to extract

```ts
export const sitePlansQuery = (siteId: number) => ({
  queryKey: [ 'site', siteId, 'plans', 'current' ],
  queryFn: () => fetchSitePlans(siteId);
})

export const sitePlansQuery = (siteId: number) => ({
  ...sitePlansQuery(siteId),
  select: ....
})
```

and then it's `sitePlansQuery()` than we `ensureQueryData`.

Reply:
Sorry -- what do you mean?

Reply:
Hmmm... you're right. Maybe we can create a util function `getCurrentPlan( plans )` 😅 

@p-jackson, what do you think?

Reply:
Or I guess we can do something like this (untested)

```ts
		if ( preload ) {
			await Promise.all( [
				queryClient.ensureQueryData( sitePlansQuery( site.ID ) ),
				...
			] );

			const currentPlan = queryClient.getQueryData( siteCurrentPlanQuery( site.ID ) );
			...
		}
```

In any case, now I don't really feel strongly haha, maybe we can iterate it later.

Reply:
Let's iterate it later 😄

Reply:
Yes I agree, let's iterate later. Looks like we might need to take a different approach to selector functions 😮